### ISSUE TITLE
feat: add support for otlp endpoints

### ISSUE THREAD

**Issue by kruskall** [elastic/apm-perf#18]
The soaktest runner sends events to the intake v2 endpoints and is able to generate load from the major agents however there's no option to send events to the otlp endpoint.

We should add support for that

**Comment by endorama** [comment]
Do we want a boolean switch (sending either Elastic APM or OTLP) or an enable switch (sending Elastic APM only or both Elastic APM and OTLP)?

In case is the first occurrence, would we prefer:
- `apmsoak run --protocol=apm|otlp`
- `apmsoak elastic-apm ` and `apmsoak otlp run`

Does `internal/soaktest` support generating OTLP already? Also it seems `apmsoak` only sends payload through HTTP, which would leave out OTLP/gRPC.

**Comment by endorama** [comment]
It make sense, but were you thinking of supporting both OTLP via HTTP and gRPC? I briefly looked at the code but it seems it is HTTP oriented so supporting gRPC would require more adjustments.

The main reason I'm thinking we need gRPC too is that if the purpose of the test is to create constant load we can observe differences between HTTP and gRPC that we are probably interested in.

**Comment by simitt** [comment]
Agree, let's start with HTTP and scope this task to OTLP over HTTP.

**Comment by endorama** [comment]
I started working on this again, following @kruskall suggestion.

**Comment by carsonip** [comment]
The initial effort in #45 using otlp/http in JSON protobuf encoding is blocked by https://github.com/elastic/apm-server/issues/12636 . It may be better to evaluate alternative tools like https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/cmd/telemetrygen

**Comment by simitt** [comment]
https://github.com/elastic/apm-perf/pull/58 addressed this partially. Support for apmsoak is still missing.
